src/scaffold.ts

```diff
@@ -92,27 +92,29 @@ export const scaffoldProjectRequest = async (templateRequestOptions?: PrefilledT
     return;
   }
 
-  if (pickedTemplate !== undefined) {
-    let itemType: string | undefined;
-    if (templateRequestOptions?.templateName) {
-      // only URIs will specify the templateName
-      itemType = "uri";
-    } else if (templateRequestOptions?.topic) {
-      // no templateName, but we have a topic name so this must've come from a topic tree item
-      itemType = "topic";
-    } else if (templateRequestOptions?.bootstrap_server) {
-      // no templateName, but we have a bootstrap_server (but no topic name) so this must've come from a Kafka cluster tree item
-      itemType = "cluster";
-    }
-    logUsage(UserEvent.ProjectScaffoldingAction, {
-      status: "template picked",
-      templateName: pickedTemplate.spec!.display_name,
-      itemType,
-    });
-  } else {
+  if (!pickedTemplate) {
+    // user canceled the quickpick
     return;
   }
 
+  let telemetrySource: string | undefined;
+  if (templateRequestOptions?.templateName) {
+    // only URIs will specify the templateName
+    telemetrySource = "uri";
+  } else if (templateRequestOptions?.topic) {
+    // no templateName, but we have a topic name so this must've come from a topic tree item
+    telemetrySource = "topic";
+  } else if (templateRequestOptions?.bootstrap_server) {
+    // no templateName, but we have a bootstrap_server (but no topic name) so this must've come from a Kafka cluster tree item
+    telemetrySource = "cluster";
+  }
+  logUsage(UserEvent.ProjectScaffoldingAction, {
```

Comment:
I assume this is for picked template via quickpick or URI... would we want to differentiate?

Reply:
That's what `itemType` sets currently -- if it's from the sidebar, it'll be either `topic` or `cluster`, and if it's from the URI handling it'll be `uri`. If `telemetrySource` is left undefined it won't be sent (and will just represent the normal "Generate Project from Template" flow)